Add custom screen object to render effects on top of the stream

levs42 opened the pull request:
## Description & motivation
* Feature: add dedicated video effects screen object

This PR introduces a new screen object that allows to render effects on top of the other screen objects. By default `IOVideoMixer` uses `videoTrackScreenObject` to render effects. As result, multicam covers the effects during the rendering. Having custom object allows to add it as a screen child and display extra elements on top of the stream. 

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Screenshots:
<details>
  <summary>Spoiler</summary>

![Screenshot 2024-08-28 at 3 15 06 PM](https://github.com/user-attachments/assets/98ca578b-8888-4a38-9acd-11f40bbaa9f3)

  
</details>

shogo4405 reviewed (changes requested):
From a maintenance perspective, I want to avoid increasing the number of ScreenObjects optimized for specific needs.

If place it on the HaishinKit side, I would like to make it more general so that other users can also utilize it easily. It can be accepted with the following changes:

Allow ImageScreenObject to be set from the CIImage library side since there is currently no CIImage-based ImageScreenObject.
Rename it to CIImageScreenObject."

levs42 commented:
I've updated the PR, please take a look. I also added an extra check to `ScreenRendererByCPU` to avoid the situation where screen object was rendered for some time, than it became invisible, but the rendered still has an image in the `images` array and renders this cached image.

shogo4405 commented:
I believe this is a class that should be placed in the app layer as a unique extension on the app side. There is no benefit in placing it on the HK side, so please position it in the app layer.

levs42 commented:
Sounds good. How about `ScreenRendererByCPU` cached image issue, is the extra check something that can be merged?

shogo4405 commented:
I cannot make a decision because there is no material (such as tests or reproducible code) to judge from. Let’s close this as a PR. When you have time, please provide reproducible code and create an issue. I will take care of the fix on my end.

From a design perspective, that check should be on the ScreenObjectContainer side, not on the ScreenRenderer side. https://github.com/shogo4405/HaishinKit.swift/blob/4adecd1a89ecde9bf873fe28911c17a0c93aa4ec/Sources/Screen/ScreenObjectContainer.swift#L60-L7

If I place it on the ScreenRenderer side, it could potentially cause other issues, or it might not. That’s why I’m being cautious.
